### ISSUE TITLE
Support the prefix under complex conditions

### ISSUE THREAD

**Issue by yisibl** [postcss/autoprefixer#511]
### Input:
```scss
p {
  column-width: 150px;
  column-rule: 2px solid blue;
  transition: column-rule 1s;
}
```

Use `"browsers": ["firefox >= 15", "ios >= 8", "android >= 4.0", "and_uc >= 9.9"]`

### Now output:

```scss
p {
  -webkit-column-width: 150px;
     -moz-column-width: 150px;
          column-width: 150px;
  -webkit-column-rule: 2px solid blue;
     -moz-column-rule: 2px solid blue;
          column-rule: 2px solid blue;
  -webkit-transition: -webkit-column-rule 1s;
     -moz-transition: -moz-column-rule 1s;
          transition: column-rule 1s;
}
```
Some browsers are not effective. [Demo1](http://jsbin.com/viyototawo/edit?html,css,output), [Demo2](http://jsbin.com/gawijotoqu/edit?html,css,output)

We need the right solution to this situation.

### I think it should be like this:

```scss
/* If only support prefix-transition */
// "browsers": ["firefox >= 15", "ios >= 8", "android >= 4.0", "and_uc >= 9.9"]
// output:

p {
  -webkit-column-width: 150px;
     -moz-column-width: 150px;
          column-width: 150px;
  -webkit-column-rule: 2px solid blue;
     -moz-column-rule: 2px solid blue;
          column-rule: 2px solid blue;
  -webkit-transition: -webkit-column-rule 1s;
     -moz-transition: -moz-column-rule 1s;
          transition: column-rule 1s;
}

/* If support no prefix transition */
// "browsers": ["firefox >= 15", "ios >= 8"]
// output:

p {
  -webkit-column-width: 150px;
     -moz-column-width: 150px;
          column-width: 150px;
  -webkit-column-rule: 2px solid blue;
     -moz-column-rule: 2px solid blue;
          column-rule: 2px solid blue;
          transition: -webkit-column-rule 1s; /* WebKit */
          transition: -moz-column-rule 1s;
          transition: column-rule 1s; /* Opera 12.1 */
          transition: -webkit-column-rule 1s, /* Blink */
                         -moz-column-rule 1s, /* Firefox */
                              column-rule 1s;
}

/* If column-* and transition remove perfix */
// output:
p {
  column-width: 150px;
  column-rule: 2px solid blue;
  transition: column-rule 1s;
}
```

**Comment by ai** [comment]
@yisibl can output be just:

```css
p {
  -webkit-column-width: 150px;
     -moz-column-width: 150px;
          column-width: 150px;
  -webkit-column-rule: 2px solid blue;
     -moz-column-rule: 2px solid blue;
          column-rule: 2px solid blue;
          transition: -webkit-column-rule 1s, /* Blink */
                         -moz-column-rule 1s, /* Firefox */
                              column-rule 1s;
}
```

**Comment by ai** [comment]
Nope, seems like I need somethings like:

```css
p {
          transition: -webkit-column-rule 1s; /* WebKit */
          transition: column-rule 1s, /* Opera 12.1 and future */
          transition: -webkit-column-rule 1s, /* Blink */
                         -moz-column-rule 1s, /* Firefox */
                              column-rule 1s;
}
```

I hate Safari.

**Comment by yisibl** [comment]
@ai  That's right.

**Comment by ai** [comment]
50% is done in `transition` branch.

I hate Safari.

**Comment by ai** [comment]
I done it. I will go get drunk tonight.

**Comment by yisibl** [comment]
@ai Nice!

**Comment by ai** [comment]
Released in 6.1
